File: 1-1-2025_Maximum Score After Splitting a String.java
class Solution {
    public int count1(String s, int j) {
        int c = 0;
        for (int i = 0; i <= j; i++) {
            if (s.charAt(i) == '0') {
                c++;
            }
        }
        return c;
    }

    public int count2(String s, int j) {
        int c = 0;
        for (int i = j; i < s.length(); i++) {
            if (s.charAt(i) == '1') {
                c++;
            }
        }
        return c;
    }

    public int maxScore(String s) {
        int max = 0;
        for (int i = 0; i < s.length() - 1; i++) {
            int score = count1(s, i) + count2(s, i + 1);
            max = Math.max(max, score);
        }
        return max;
    }
}

File: 1-1-2025_Maximum Score After Splitting a String.py
class Solution:
    def count1(self, s, j):
        c = 0
        for i in range(j + 1):
            if s[i] == '0':
                c += 1
        return c
    
    def count2(self, s, j):
        c = 0
        for i in range(j, len(s)):
            if s[i] == '1':
                c += 1
        return c
    
    def maxScore(self, s):
        max_score = 0
        for i in range(len(s) - 1):
            score = self.count1(s, i) + self.count2(s, i + 1)
            max_score = max(max_score, score)
        return max_score